OpenRA.Game/Network/Connection.cs

```diff
@@ -30,12 +34,48 @@ public interface IConnection : IDisposable
 	{
 		int LocalClientId { get; }
 		ConnectionState ConnectionState { get; }
+		IPEndPoint EndPoint { get; }
 		void Send(int frame, List<byte[]> orders);
 		void SendImmediate(IEnumerable<byte[]> orders);
 		void SendSync(int frame, byte[] syncData);
 		void Receive(Action<int, byte[]> packetFn);
 	}
 
+	public class ConnectionTarget
+	{
+		readonly DnsEndPoint[] endpoints;
+
+		public ConnectionTarget()
+		{
+			endpoints = new DnsEndPoint[] { new DnsEndPoint("invalid", 0) };
+		}
+
+		public ConnectionTarget(string host, int port)
+		{
+			endpoints = new DnsEndPoint[] { new DnsEndPoint(host, port) };
+		}
+
+		public ConnectionTarget(IEnumerable<DnsEndPoint> endpoints)
+		{
+			this.endpoints = endpoints.ToArray();
+		}
+
+		public IEnumerable<IPEndPoint> GetConnectEndPoints()
+		{
+			return endpoints
+				.SelectMany(e => Dns.GetHostAddresses(e.Host)
```

Comment:
This crashes if a bogus address is entered into the direct connect dialog.
```
Exception of type `System.Net.Sockets.SocketException`: Could not resolve host 'fail'
  at System.Net.Dns.Error_11001 (System.String hostName) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-10/external/bockbuild/builds/mono-x64/mcs/class/System/System.Net/Dns.cs:308 
  at System.Net.Dns.GetHostByName (System.String hostName) [0x00021] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-10/external/bockbuild/builds/mono-x64/mcs/class/System/System.Net/Dns.cs:441 
  at System.Net.Dns.GetHostEntry (System.String hostNameOrAddress) [0x00052] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-10/external/bockbuild/builds/mono-x64/mcs/class/System/System.Net/Dns.cs:397 
  at System.Net.Dns.GetHostAddresses (System.String hostNameOrAddress) [0x00056] in /Users/builder/jenkins/workspace/build-package-osx-mono/2019-10/external/bockbuild/builds/mono-x64/mcs/class/System/System.Net/Dns.cs:424 
  at OpenRA.Network.ConnectionTarget+<>c.<GetConnectEndPoints>b__4_0 (System.Net.DnsEndPoint e) [0x0000d] in /Users/paul/src/OpenRA/OpenRA.Game/Network/Connection.cs:66 
...
```

Reply:
That method also has another issue: If the name resolves to only IPv6 addresses and the system "does not have IPv6 installed" (whatever that means), then it will return an empty list which will run into a timeout later.

Reply:
Exceptions are now ignored and a completely empty result will lead to a proper error later.